### PR TITLE
enabled external persons to participate in a thesis proposal THS-33 #resolve

### PR DESCRIPTION
**Pull request by pedrorijo91**

### CODE REVIEW

File: src/main/dml/fenixedu-academic-thesis.dml [hunk 1]
```diff
@@ -141,11 +146,21 @@ relation ThesisProposalConfiguration {
...
+    }
+}
+
+relation ThesisProposalParticipantExternalUser {
+    public ExternalUser playsRole externalUser {
+    multiplicity 0..1;
+    }
+
+    public ThesisProposalParticipant playsRole thesisProposalParticipant {
+    multiplicity 0..*;
```
Comment (pedrosan7os) [inline src/main/dml/fenixedu-academic-thesis.dml]: The same ExternalUser object is to be used in several participations?

File: src/main/java/org/fenixedu/academic/thesis/domain/ThesisProposalParticipant.java [hunk 2]
```diff
@@ -54,6 +67,8 @@ public void delete() {
         this.setUser(null);
         this.setThesisProposal(null);
 
+        this.setExternalUser(null);
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/academic/thesis/domain/ThesisProposalParticipant.java]: If ExternalUser instances are not shared they should be deleted when the participation is deleted

File: src/main/java/org/fenixedu/academic/thesis/domain/ThesisProposalParticipant.java [hunk 3]
```diff
@@ -61,4 +76,12 @@ public void delete() {
     protected void checkForDeletionBlockers(Collection<String> blockers) {
         super.checkForDeletionBlockers(blockers);
     }
+
+    public String getName() {
+        if (getUser() != null) {
+            return getUser().getName();
+        } else {
+            return getExternalUser().getName() + " " + getExternalUser().getEmail();
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/academic/thesis/domain/ThesisProposalParticipant.java:84]: maybe some parenthesis around the email

Comment (pedrorijo91) [inline src/main/java/org/fenixedu/academic/thesis/domain/ThesisProposalParticipant.java:84]: parenthesis are already used for participation percentage..

**Pull request merged**